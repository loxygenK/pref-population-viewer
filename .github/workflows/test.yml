name: "Test"

on:
  pull_request:
    types:
      - opened
      - synchronize

jobs:
  check:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v3

      - uses: actions/cache@v2
        with:
          path: ~/.pnpm-store
          key: ${{ runner.os }}-pnpm_store-${{ hashFiles('**/pnpm-lock.yaml') }}
          restore-keys: |
            ${{ runner.os }}-pnpm_store-

      - uses: pnpm/action-setup@v2
        with:
          version: 6.32
          run_install: true

      - run: pnpm test

      - uses: actions/upload-artifact@v3
        if: failure()
        with:
          name: failed-test-recording
          path: |
            ./cypress/videos
            ./cypress/screenshots
