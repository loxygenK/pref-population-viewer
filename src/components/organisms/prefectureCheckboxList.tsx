import React from "react";
import useSWR from "swr";
import { useAPIProxyClient } from "~/api/proxy/hook/useAPIProxyClient";
import { Prefecture } from "~/domain/prefecture";
import { ValueWithID } from "~/types/valueWithId";
import { CheckBoxList } from "../molecules/checkboxList";

const usePrefectures = () => {
  const apiProxyClient = useAPIProxyClient();
  return useSWR<Prefecture[]>("prefecture", () =>
    apiProxyClient.pref.fetchPrefectures()
  );
};

export interface PrefectureCheckBoxListProps {
  selectedPrefs: Prefecture[];
  onChange: (prefs: Prefecture[]) => void;
}
export const PrefectureCheckBoxList: React.FC<PrefectureCheckBoxListProps> = ({
  selectedPrefs,
  onChange,
}) => {
  const { data: prefs, error } = usePrefectures();

  const checkboxValues: Array<ValueWithID<string>> | undefined =
    React.useMemo(() => {
      return prefs?.map((p) => ({ id: p.id, value: p.name }));
    }, [prefs]);

  const selectedPrefIDs: Array<string> = React.useMemo(() => {
    return selectedPrefs.map((p) => p.id);
  }, [selectedPrefs]);

  const onCheckBoxChanged = (ids: string[]) => {
    if (prefs === undefined) {
      return;
    }

    const resolvedPrefs = prefs.filter((p) => ids.includes(p.id));
    onChange(resolvedPrefs);
  };

  if (checkboxValues === undefined) {
    // TODO: Create more fancier loading screen
    return <div>Loading...</div>;
  }

  return (
    <CheckBoxList
      values={checkboxValues}
      checkedIDs={selectedPrefIDs}
      onChange={onCheckBoxChanged}
    />
  );
};
