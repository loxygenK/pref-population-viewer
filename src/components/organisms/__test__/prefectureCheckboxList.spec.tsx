import { renderStatefulComponent } from "~/test/renderStatefulComponent";
import { PrefectureCheckBoxList } from "../prefectureCheckboxList";

describe("Header component", () => {
  it("renders correctly", async () => {
    const rendered = await renderStatefulComponent((root) => {
      root.render(
        <PrefectureCheckBoxList selectedPrefs={[]} onChange={() => {}} />
      );
    });

    expect(rendered).toMatchSnapshot();
  });
});
