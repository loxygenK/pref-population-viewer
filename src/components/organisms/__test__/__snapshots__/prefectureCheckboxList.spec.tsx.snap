// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `"<div class=\\"checkbox_list_wrapper\\"><button class=\\"toggle_button pressed show_toggle_button\\">折りたたむ</button><ul class=\\"checkbox_list \\"><li><label for=\\"check-pref-0\\"><input id=\\"check-pref-0\\" type=\\"checkbox\\">Pref. #0</label></li><li><label for=\\"check-pref-1\\"><input id=\\"check-pref-1\\" type=\\"checkbox\\">Pref. #1</label></li><li><label for=\\"check-pref-2\\"><input id=\\"check-pref-2\\" type=\\"checkbox\\">Pref. #2</label></li><li><label for=\\"check-pref-3\\"><input id=\\"check-pref-3\\" type=\\"checkbox\\">Pref. #3</label></li><li><label for=\\"check-pref-4\\"><input id=\\"check-pref-4\\" type=\\"checkbox\\">Pref. #4</label></li><li><label for=\\"check-pref-5\\"><input id=\\"check-pref-5\\" type=\\"checkbox\\">Pref. #5</label></li><li><label for=\\"check-pref-6\\"><input id=\\"check-pref-6\\" type=\\"checkbox\\">Pref. #6</label></li><li><label for=\\"check-pref-7\\"><input id=\\"check-pref-7\\" type=\\"checkbox\\">Pref. #7</label></li><li><label for=\\"check-pref-8\\"><input id=\\"check-pref-8\\" type=\\"checkbox\\">Pref. #8</label></li><li><label for=\\"check-pref-9\\"><input id=\\"check-pref-9\\" type=\\"checkbox\\">Pref. #9</label></li><li><label for=\\"check-pref-10\\"><input id=\\"check-pref-10\\" type=\\"checkbox\\">Pref. #10</label></li><li><label for=\\"check-pref-11\\"><input id=\\"check-pref-11\\" type=\\"checkbox\\">Pref. #11</label></li><li><label for=\\"check-pref-12\\"><input id=\\"check-pref-12\\" type=\\"checkbox\\">Pref. #12</label></li><li><label for=\\"check-pref-13\\"><input id=\\"check-pref-13\\" type=\\"checkbox\\">Pref. #13</label></li><li><label for=\\"check-pref-14\\"><input id=\\"check-pref-14\\" type=\\"checkbox\\">Pref. #14</label></li><li><label for=\\"check-pref-15\\"><input id=\\"check-pref-15\\" type=\\"checkbox\\">Pref. #15</label></li><li><label for=\\"check-pref-16\\"><input id=\\"check-pref-16\\" type=\\"checkbox\\">Pref. #16</label></li><li><label for=\\"check-pref-17\\"><input id=\\"check-pref-17\\" type=\\"checkbox\\">Pref. #17</label></li><li><label for=\\"check-pref-18\\"><input id=\\"check-pref-18\\" type=\\"checkbox\\">Pref. #18</label></li><li><label for=\\"check-pref-19\\"><input id=\\"check-pref-19\\" type=\\"checkbox\\">Pref. #19</label></li><li><label for=\\"check-pref-20\\"><input id=\\"check-pref-20\\" type=\\"checkbox\\">Pref. #20</label></li><li><label for=\\"check-pref-21\\"><input id=\\"check-pref-21\\" type=\\"checkbox\\">Pref. #21</label></li><li><label for=\\"check-pref-22\\"><input id=\\"check-pref-22\\" type=\\"checkbox\\">Pref. #22</label></li><li><label for=\\"check-pref-23\\"><input id=\\"check-pref-23\\" type=\\"checkbox\\">Pref. #23</label></li><li><label for=\\"check-pref-24\\"><input id=\\"check-pref-24\\" type=\\"checkbox\\">Pref. #24</label></li><li><label for=\\"check-pref-25\\"><input id=\\"check-pref-25\\" type=\\"checkbox\\">Pref. #25</label></li><li><label for=\\"check-pref-26\\"><input id=\\"check-pref-26\\" type=\\"checkbox\\">Pref. #26</label></li><li><label for=\\"check-pref-27\\"><input id=\\"check-pref-27\\" type=\\"checkbox\\">Pref. #27</label></li><li><label for=\\"check-pref-28\\"><input id=\\"check-pref-28\\" type=\\"checkbox\\">Pref. #28</label></li><li><label for=\\"check-pref-29\\"><input id=\\"check-pref-29\\" type=\\"checkbox\\">Pref. #29</label></li><li><label for=\\"check-pref-30\\"><input id=\\"check-pref-30\\" type=\\"checkbox\\">Pref. #30</label></li><li><label for=\\"check-pref-31\\"><input id=\\"check-pref-31\\" type=\\"checkbox\\">Pref. #31</label></li><li><label for=\\"check-pref-32\\"><input id=\\"check-pref-32\\" type=\\"checkbox\\">Pref. #32</label></li><li><label for=\\"check-pref-33\\"><input id=\\"check-pref-33\\" type=\\"checkbox\\">Pref. #33</label></li><li><label for=\\"check-pref-34\\"><input id=\\"check-pref-34\\" type=\\"checkbox\\">Pref. #34</label></li><li><label for=\\"check-pref-35\\"><input id=\\"check-pref-35\\" type=\\"checkbox\\">Pref. #35</label></li><li><label for=\\"check-pref-36\\"><input id=\\"check-pref-36\\" type=\\"checkbox\\">Pref. #36</label></li><li><label for=\\"check-pref-37\\"><input id=\\"check-pref-37\\" type=\\"checkbox\\">Pref. #37</label></li><li><label for=\\"check-pref-38\\"><input id=\\"check-pref-38\\" type=\\"checkbox\\">Pref. #38</label></li><li><label for=\\"check-pref-39\\"><input id=\\"check-pref-39\\" type=\\"checkbox\\">Pref. #39</label></li><li><label for=\\"check-pref-40\\"><input id=\\"check-pref-40\\" type=\\"checkbox\\">Pref. #40</label></li><li><label for=\\"check-pref-41\\"><input id=\\"check-pref-41\\" type=\\"checkbox\\">Pref. #41</label></li><li><label for=\\"check-pref-42\\"><input id=\\"check-pref-42\\" type=\\"checkbox\\">Pref. #42</label></li><li><label for=\\"check-pref-43\\"><input id=\\"check-pref-43\\" type=\\"checkbox\\">Pref. #43</label></li><li><label for=\\"check-pref-44\\"><input id=\\"check-pref-44\\" type=\\"checkbox\\">Pref. #44</label></li><li><label for=\\"check-pref-45\\"><input id=\\"check-pref-45\\" type=\\"checkbox\\">Pref. #45</label></li><li><label for=\\"check-pref-46\\"><input id=\\"check-pref-46\\" type=\\"checkbox\\">Pref. #46</label></li><li><label for=\\"check-pref-47\\"><input id=\\"check-pref-47\\" type=\\"checkbox\\">Pref. #47</label></li><li><label for=\\"check-pref-48\\"><input id=\\"check-pref-48\\" type=\\"checkbox\\">Pref. #48</label></li><li><label for=\\"check-pref-49\\"><input id=\\"check-pref-49\\" type=\\"checkbox\\">Pref. #49</label></li></ul></div>"`;
